### PR TITLE
Update HTTPS

### PR DESCRIPTION
**Pull request by RailonA**
Generator-Roosevelt creates a sample application that runs on HTTPS.

### CODE REVIEW

File: generators/app/promptingHelpers.js [hunk 1]
```diff
@@ -37,12 +37,12 @@ helper.sanitizePackageName = function (appName) {
...
-    return 'Incorrect input please enter in this format (e.g. US, CA)'
-  }
-  return true
-}
+// helper.countryValidation = function (input) {
+//   if (!/^[A-Z]{2}$/.test(input)) {
+//     return 'Incorrect input please enter in this format (e.g. US, CA)'
+//   }
+//   return true
+// }
```
Comment (kethinov) [inline generators/app/promptingHelpers.js:45]: why remove this?

if it's not useful / not used, we should just delete it, not comment it out

File: generators/app/templates/mvc/models/teddy/homepage.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = (model) => {
-  model.content.hello = 'Hi! I\'m a variable trickling down through the MVC structure!'
+  model.content.hello = 'HTTPS CHANGES'
```
Comment (kethinov) [inline generators/app/templates/mvc/models/teddy/homepage.js:2]: revert this

**Review by kethinov — CHANGES_REQUESTED** [review]

File: generators/app/index.js [hunk 3]
```diff
@@ -4,6 +4,18 @@ const defaults = require('./templates/defaults.json')
 const beautify = require('gulp-beautify')
 const filter = require('gulp-filter')
 const terminalLink = require('terminal-link')
+const selfsigned = require('selfsigned');
```
Comment (kethinov) [inline generators/app/index.js:7]: this needs to be added to package.json and you need to commit a fresh package-lock

**Review by kethinov — CHANGES_REQUESTED** [review]

**Review by kethinov — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -23,6 +23,7 @@
     "gulp-beautify": "~3.0.0",
     "gulp-filter": "~7.0.0",
     "node-forge": "~0.10.0",
+    "selfsigned": "^2.0.1",
```
Comment (kethinov) [inline package.json:26]:
```suggestion
    "selfsigned": "~2.0.1",
```

Comment (kethinov) [reply package.json:26]: also make sure to run `standard --fix` before committing, not sure this will pass the linter

**Review by kethinov — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (kethinov) [reply package.json:26]: don't forget to merge the suggestion i put above

some more things to do:

alter this:

```
To run the app:
- Install dependencies: npm i
- To run in dev mode:   npm run dev
- To run in prod mode:  npm run prod
Once running, visit http://localhost:10709
```

The link should include the HTTPS link, not the HTTP link.

also the HTTP server should be disabled. Make it just HTTPS.

and randomize the port of the HTTPS server so it's not always 43733. there is an example of how to do a random port in the HTTP code. just reuse that

File: generators/app/index.js [hunk 5]
```diff
@@ -95,313 +91,6 @@ module.exports = class extends Generator {
       })
   }
 
-  spa () {
```
Comment (kethinov) [inline generators/app/index.js:98]: pretty sure you deleted way more code here than you should have

we very much intend to continue supporting the SPA feature

you should test it to ensure it still works

**Review by kethinov — CHANGES_REQUESTED** [review]

File: generators/app/index.js [hunk 6]
```diff
@@ -24,21 +35,6 @@ module.exports = class extends Generator {
       required: false,
       desc: 'Skips all prompts and creates a Roosevelt app with all defaults.'
     })
-
-    this.option('install-deps', {
```
Comment (Autre31415) [inline generators/app/index.js:28]: Why are these options being deleted?

**Review by kethinov — APPROVED** [review]

**Pull request merged**